File: Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* mynode=head;
        ListNode* nxtnode=head;
        if (head==NULL){
            return head;
        } 
        else
        {while (mynode->next!=NULL ){
           // nxtnode=mynode->next;
            nxtnode=mynode->next;
            if (mynode->val==nxtnode->val){
                mynode->next=nxtnode->next;
               // delete nxtnode;
                cout<<"h";
            }
             if (mynode->val!=nxtnode->val){
                 mynode=mynode->next;
                 cout<<"i";
             }
        }
}
        
   return head; }
};

File: remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        arr=[]
        temp=head
        prev=None
        while temp:
            if temp.val not in arr:
                arr.append(temp.val)
                print(arr)
                prev=temp
            elif temp.val in arr:
                prev.next=temp.next
            temp=temp.next
        return head